hypothesis-python/src/hypothesis/strategies/_internal/strategies.py

```diff
@@ -531,18 +531,9 @@ def do_draw(self, data):
         if isinstance(result, SearchStrategy) and all(
             isinstance(x, SearchStrategy) for x in self.elements
         ):
-            max_num_strats = 3
-            preamble = (
-                f"(first {max_num_strats}) "
-                if len(self.elements) > max_num_strats
-                else ""
-            )
-            strat_reprs = ", ".join(
-                map(get_pretty_function_description, self.elements[:max_num_strats])
-            )
             data._sampled_from_all_strategies_elements_message = (
                 "sample_from was given a collection of strategies: "
-                f"{preamble}{strat_reprs}. Was one_of intended?"
+                "{!r}. Was one_of intended?", self.elements
```

Comment:
The warning triggered on this code - on win only, for some reason - so I delayed the repr until we actually add the note.

Since it caused a test failure, I propose that this does not violate the one-thing-per-PR rule ;-)

Reply:
Note - max 3 is removed, since I think it was only added as a workaround for this specific problem in the first place

Reply:
(not really outdated, I just rebased for the RELEASE.rst conflict)